MITADS/opensubtitles_exporter.py

```diff
@@ -0,0 +1,354 @@
+  [ re.compile('(\s|^)(\')([^\']*)(\')(\s|$)'), r'\3' ],
+
+  ## punteggiatura non significativa alla fine del testo  
+  [ re.compile('(.*)(\.)$'), r'\2' ],
+
+  ##normalizziamo vovcali accentate
+  [ re.compile('a\'(\s|$|,|\.|\?)'), r'à\1' ],
+  [ re.compile('e\'(\s|$|,|\.|\?)'), r'è\1' ],
+  [ re.compile('i\'(\s|$|,|\.|\?)'), r'ì\1' ],
+  [ re.compile('o\'(\s|$|,|\.|\?)'), r'ò\1' ],
+  [ re.compile('u\'(\s|$|,|\.|\?)'), r'ù\1' ],
+
+  ## cancelletto con numero , presente nei titoli di apertura   
+  [ re.compile('#\d+'), u'' ],
+  [ re.compile('#|\s°'), u'' ], 
+
+  ##alcune normalizzazioni con numeri. sono presenti, vedere se serve questo tipo di normalizzazione
+  ##ISSUE: potrebbero esserci problemi con maschile e femminile, vedi 6° strada ad esempio
+  [ u'1°' , u'primo' ],
+  [ u'2°' , u'secondo' ],
+  [ u'3°' , u'terzo' ],
+  [ u'4°' , u'quarto' ],
+  [ u'5°' , u'quinto' ],
+  [ u'49°' , u'quarantanovesimo' ],
+  
+  ##numero, metto uno spazio perchè certe volte il simbolo è attaccato al numero
+  [ u'n°' , u'numero ' ],  
+
+  ##importi in valuta, sono presenti   
+  [ re.compile('\$\s*([0-9]+[.,]{0,1}[0-9]*)'), r'\1 dollari' ], 
+  [ re.compile('([0-9]+[.,]{0,1}[0-9]*)\s*\$'), r'\1 dollari' ],
+  [ re.compile('(₤|£)\s*([0-9]+[.,]{0,1}[0-9]*)'), r'\2 lire' ], 
+  [ re.compile('([0-9]+[.,]{0,1}[0-9]*)\s*₤'), r'\1 lire' ],
+  [ re.compile('(€)\s*([0-9]+[.,]{0,1}[0-9]*)'), r'\2 euro' ], 
+  [ re.compile('([0-9]+[.,]{0,1}[0-9]*)\s*€'), r'\1 euro' ],
+  ##trim spazi
+  [ re.compile('\s+'), u' ' ]
+]
+
+
+def maybe_normalize(value, mapping=mapping_normalization):
+  for norm in mapping:
+    if type(norm[0]) == str:
+      value = value.replace(norm[0], norm[1])
+    elif isinstance(norm[0], Pattern):
+      try:
+        value = norm[0].sub(norm[1], value)
+      except Exception as e:
+        raise(e)
+    else:
+      print('UNEXPECTED', type(norm[0]), norm[0])
+
+  ##remove markup tags...(non sono presenti nel dataset )
+  #value = loads(value)
+
+  return value
+
+def line_not_relevant(text):
+  ##implementare logiche di esclusione righe
+  if(len(text)<2):
+    return True
+
+  return False
+
+
+def clear_text(text):
+  ##text = re.sub(r'[^\x00-\x7F]+',' ', text)  mi toglie anche le È
+
+  #PER I CARATTERI UNICODE ho usato libreria unidecode, questi non servono più
+  #text = text.replace('\u200b',' ')## ZERO WIDTH SPACE
+  #text = text.replace(u'ε','e') ##GREEK SMALL LETTER EPSILON
+  #text = text.replace(u'♪','') ## char ♪
+  #text = text.replace(u'♫','') ## char ♫
+  #text = text.replace(u'¶','') ## char ♫
+  
+  #text = text.sub(r'[^\x00-\x7F]+',' ', text_ret)
+  
+  text = text.strip()
+
+  return text
+
+def normalize_text(text):
+
+  ##SE E SOLO SE  tutto il testo è tutto upper case allora lo porto a lower case
+  if(text.isupper()):
+    text = text.lower()
+
+  ##se una sentence contiene una sola parola con caratteri speciali 
+  # ALLORA tutta la sentence viene esclusa (fa  parte dei titoli di testa)
```

Comment:
If the line starts with one of those symbols we ignore because is part of the opening